### DISCUSSION TITLE
Access MONAI Label app in remote server with authentication through 3D Slicer

### DISCUSSION THREAD

**Discussion by VarshaTolani** [Project-MONAI/MONAILabel#1692] [Q&A] [answered]
Hey there!

I am trying to access the MONAI Label app, which is running on a remote server (Jupyter environment in OpenShift), from 3D Slicer. To my understanding, adding the IP and port of the remote server should be enough to establish the connection, but my requests are redirected through a proxy and I have some enabled hosts locally, so I need to use the server URL directly (instead of IP).

The issue is, to access the OpenShift server I need to authenticate with my user, I have tried to do so with the "https://user:password@domain" URL format, but it is not working.

Is there any way I can connect to a remote server with authentication from 3DSlicer or MONAI's extension?

Any help would be appreciated!

Thanks :)

**Comment by lassoan** [comment]
In the Slicer module, it would be easy to add a textbox where you could specify custom information that would be added to the request header. This extra header information, could include an access token.

**Reply by VarshaTolani** [reply]
As this is an open source code, are you suggesting that I update the code by myself and make a PR to the repository? I could be able to do that, but I guess it would require some time before the PR gets approved so that I can re-install the extension.
Or is there any other workaround you suggest?

Thanks for the message anyways, I'll start by taking a look at the module script!

**Reply by lassoan** [reply, marked as answer]
I would expect that you need to make a few minor changes, so review and merge of a pull request should not take long. But you can modify the Python files on your computer and use the updated files right away - you don't need to wait for your pull request to be merged.

**Reply by VarshaTolani** [reply]
I am trying to do that just now. I'll be testing to make sure it works so that I can upload the changes, hopefully it helps others as well.
Thank you! 😄
